arrow-array/src/array/byte_array.rs

```diff
@@ -304,6 +304,10 @@ impl<T: ByteArrayType> GenericByteArray<T> {
     }
 
     /// Returns the element at index `i`
+    ///
+    /// Note: This method does not check for nulls and the value is arbitrary
+    /// (but still valid) if [`is_null`](Self::is_null) returns true for the index.
```

Comment:
Since the opposite of `is_null` is `is_valid`, I'd call this well-defined, not `valid`

```suggestion
    /// (but still well-defined) if [`is_null`](Self::is_null) returns true for the index.
```